Clips-Player-Sample-Project/clips-example/VideoPlayerView.swift

```diff
@@ -0,0 +1,276 @@
+//
+//  VideoPlayerView.swift
+//  Clips-Example
+//
+//  Created by Alex on 11.3.2022.
+//  Copyright © 2022 GIPHY. All rights reserved.
+//
+
+import UIKit
+import AVFoundation
+
+/*
+This file represents sample code for a simple video player view, complete with looping capabilities, a sound control, and a "shimmering" animation to indicate the video loading state. These components are intended to be generic and independent of the SDK.
+
+ Learn more about Clips (GIFs with Sound!) in the clips.md file.
+  
+*/
+
+import Foundation
+import AVFoundation
+
+@objcMembers
+public class VideoPlayerView: UIView {
+    
+    private var cache: URLCache = {
+        let cache = URLCache()
+        cache.diskCapacity = 1000 * 1000 * 1000
+        cache.memoryCapacity = 1000 * 1000 * 1000
+        return cache
+    }()
+    
+    let lock = NSRecursiveLock()
```

Comment:
@cgmaier 
Yes, that's correct. 
These functions must be called in the main thread, it will ensure they run synchronously.
But we had issues in the ReactNative context where the renderer uses three different threads.
That's why we keep this lock mechanism to add one more protection layer.